src/extension/module.js

```diff
@@ -1927,126 +1965,102 @@
-          }
-          // check if button is enabled
-          if (typeof plugin.button.isEnabled === 'function' && !plugin.button.isEnabled(this)) {
-            $button.setAttribute('disabled', '');
-          }
-          // check if button is pressed
-          if (typeof plugin.button.isPressed === 'function' && plugin.button.isPressed(this)) {
-            $button.classList.add('pressed');
-            $button.removeAttribute('tabindex');
-          }
-          // fire event when plugin button is clicked
-          $button.addEventListener('click', () => fireEvent(this, 'pluginused', {
-            id: plugin.id,
-            button: $button,
-          }));
+      } else if (!plugin.isShown) {
+        // remove existing plugin
+        $plugin.remove();
+        return null;
+      } else if (plugin.override) {
+        // replace existing plugin
+        const $existingPlugin = $plugin;
+        $plugin = appendTag($existingPlugin.parentElement, getPluginCfg(plugin), $existingPlugin);
+        $existingPlugin.remove();
+      }
+      // add elements
+      if (plugin.elements && Array.isArray(plugin.elements)) {
+        plugin.elements.forEach((elem) => appendTag($plugin, elem));
+      }
+      // add or update button
+      if (plugin.button) {
+        plugin.button.action = plugin.button.action || (() => {});
+        const buttonCfg = {
+          tag: 'button',
+          text: plugin.button.text,
+          lstnrs: {
+            click: (e) => plugin.button.action(e, this),
+            auxclick: (e) => plugin.button.action(e, this),
+          },
+        };
+        let $button = $plugin ? $plugin.querySelector(':scope button') : null;
+        if ($button) {
+          // extend existing button
+          extendTag($button, buttonCfg);
+        } else {
+          // add button
+          $button = appendTag($plugin, buttonCfg);
         }
-        // check advanced mode
-        if (plugin.advanced && typeof plugin.advanced === 'function' && plugin.advanced(this)) {
-          $plugin.classList.add('hlx-sk-advanced-only');
+        // check if button is enabled
+        if (typeof plugin.button.isEnabled === 'function' && !plugin.button.isEnabled(this)) {
+          $button.setAttribute('disabled', '');
         }
-        if (typeof plugin.callback === 'function') {
-          plugin.callback(this, $plugin);
+        // check if button is pressed
+        if (typeof plugin.button.isPressed === 'function' && plugin.button.isPressed(this)) {
+          $button.classList.add('pressed');
+          $button.removeAttribute('tabindex');
         }
-        return $plugin;
+        // fire event when plugin button is clicked
+        $button.addEventListener('click', () => fireEvent(this, 'pluginused', {
+          id: plugin.id,
+          button: $button,
+        }));
       }
-      return null;
+      // check advanced mode
+      if (plugin.advanced && typeof plugin.advanced === 'function' && plugin.advanced(this)) {
```

Comment:
typeof undefined is undefined:

```
> plugin = {};
{}
> typeof plugin.advanced
'undefined'
```

```suggestion
      if (typeof plugin.advanced === 'function' && plugin.advanced(this)) {
```